designer/client/component-create.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import ComponentTypeEdit from "./component-type-edit";
+import { clone, ComponentTypes } from "@xgovformbuilder/model";
+import { DataContext } from "./context";
+
+/**
+ * @deprecated (keeping until tests are refactored)
+ */
+class ComponentCreate extends React.Component {
+  static contextType = DataContext;
+  state = {
+    isSaving: false,
+  };
+
+  async componentDidMount() {
+    this.setState({ name: nanoid(6) });
+  }
+
+  async onSubmit(e) {
+    e.preventDefault();
+
+    if (this.state.isSaving) {
+      return;
+    }
+
+    this.setState({ isSaving: true });
+
+    const { page } = this.props;
+    const { data, save } = this.context;
+    const { component } = this.state;
+    const copy = clone(data);
+    const updated = copy.addComponent(page.path, component);
+    await save(updated);
+    this.setState({ isSaving: false });
+  }
+
+  storeComponent = (component) => {
+    this.setState({ component });
+  };
+
+  render() {
+    const { page, allowedTypes = ComponentTypes } = this.props;
+    const { isSaving } = this.state;
+
+    return (
+      <div>
+        <form onSubmit={(e) => this.onSubmit(e)} autoComplete="off">
```

Comment:
could just be onSubmit={this.onSubmit}, will save the function being created each time?